### ISSUE TITLE
Further Firedrake installation errors

### ISSUE THREAD

**Issue by jaredcworkman** [firedrakeproject/firedrake#1613]
Hello again.

I recently tried to re-install firedrake.  I installed it a week or so ago and it worked correctly.  I took the following steps:

1:Update cmake to version 3.15.5
2:Install and source intel compilers, mpi libraries, and set proper environmental variables - MPICC=mpiicc
MPIF77=mpiifort
MPIF90=mpif90
MPICXX=mpiicpc

I check CC, FC, and CXX point to ifort, icc, and icpc

3:Update python and its headers to 3.7

Then I run
 python firedrake-install --no-package-manager
 where python is version 3.7 something

This worked around 1.5 weeks ago, I ran make alltests and it had a 95% pass rate, I also supplied a user function which calls on firedrake and uses mpirun.  It passed although it seemed to require a tremendous amount of ram to do so.

The outline of my install script is essentially  pushd ${PROGRAM_PATH}
  curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-install
  aws s3 cp  s3://rescale-software-archive/${S3_DIRECTORY}/firedrake.exp .
  #install firedrake
  chmod -R 777 *
  ./firedrake.exp
  #source their environmental script
  . $PROGRAM_PATH/firedrake/bin/activate

where the autoexpect script runs the python firedrake-install script with two (yes) commands for downloading as needed during the install.

Today I tried the EXACT same thing and on my first pass it complained of missing some library that googled mentioned was due to a bad gcc install.  I installed sudo yum install gcc64*, re-ran the install and it then complained that my c compiler was not able to make executables.

So, I reran the code using the following 
python firedrake-install --no-package-manager  --mpif90 mpiifort  --mpicxx $MPICXX --mpicc $MPICC --mpiexec mpirun
We use intel mpi so I call mpiexec mpirun, a sym link would work as well?

Then, the install proceeds until I get hung at 
###############################
-02-20 23:56:40,499 DEBUG  Popping path '/program/firedrake/firedrake/src/libspatialindex'
2020-02-20 23:56:40,499 DEBUG  New path '/program/firedrake/firedrake/src'
2020-02-20 23:56:40,500 INFO   Installing libsupermesh
2020-02-20 23:56:40,500 INFO   Cloning libsupermesh

2020-02-20 23:56:40,500 DEBUG  Running command 'git clone -q --recursive git@bitbucket.org:libsupermesh/libsupermesh.git'
###############################
When I try to execute that command by hand I get 
#################################
[root@ip-10-23-15-88 /program/firedrake]$ git clone -q --recursive git@bitbucket.org:libsupermesh/libsupermesh.git
Warning: Permanently added the RSA host key for IP address '18.205.93.1' to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
##################################
I've gone through your recent git history changes and see nothing obvious to cause this error bt I do note several recent changes since my last install.  
Attached are the config and install log
[configure.log](https://github.com/firedrakeproject/firedrake/files/4233868/configure.log)
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/4233869/firedrake-install.log)
I need to reinstall firedrake with the current version.  Can you offer any suggestions?  It seems the code base must have changed as the scripts I set up are automated and static.

**Comment by jaredcworkman** [comment]
In this post I will attach the working configure and install scripts from my install a weekish ago.
[configure.log](https://github.com/firedrakeproject/firedrake/files/4233882/configure.log)
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/4233883/firedrake-install.log)

I think something in either the permissions for libsupermesh or the way gcc is used must have changed.  I did not use an autoexpect script when I ran into these error.  All the autoexpect script does is to call a spawn on the above install commands and insert a "yes" where needed insofar as accessing remote repositories.

Thanks.
Jared
